server/auth/sso/sso.go

```diff
@@ -180,10 +180,11 @@ func newSso(
 const stateCookieName = "oauthState"
 
 func (s *sso) HandleRedirect(w http.ResponseWriter, r *http.Request) {
+	redirectUrl := r.URL.Query().Get("redirect")
 	state := pkgrand.RandString(10)
 	http.SetCookie(w, &http.Cookie{
-		Name:     stateCookieName,
-		Value:    state,
+		Name:     state,
```

Comment:
https://auth0.com/docs/protocols/state-parameters#use-the-stored-url-to-redirect-users recommends using the nonce/state as the name of the cookie for added security